pkg/operator/minio/controller.go

```diff
@@ -0,0 +1,304 @@
+	var env []v1.EnvVar
+	for k, v := range envVars {
+		env = append(env, v1.EnvVar{Name: k, Value: v})
+	}
+
+	podSpec := v1.PodTemplateSpec{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    map[string]string{"app": minioLabel},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  ctrName,
+					Image: ctrImage,
+					Env:   env,
+					Ports: []v1.ContainerPort{{ContainerPort: port}},
+					// TODO: Let's go about building these args in a more elegant way.
+					Args: c.buildMinioCtrArgs(name, name, namespace, numServers, objectStoreDataDir),
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      minioVolumeName,
+							MountPath: objectStoreDataDir,
+						},
+					},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: minioVolumeName,
+					VolumeSource: v1.VolumeSource{
+						PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+							ClaimName: minioPVCName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return podSpec
+}
+
+func (c *MinioController) makeMinioStatefulSet(name, namespace string, spec miniov1alpha1.ObjectStoreSpec) (*v1beta2.StatefulSet, error) {
+	appsClient := c.context.Clientset.AppsV1beta2()
+
+	envVars := map[string]string{
+		"MINIO_ACCESS_KEY": spec.Credentials.AccessKey,
+		"MINIO_SECRET_KEY": spec.Credentials.SecretKey}
+
+	podSpec := c.makeMinioPodSpec(name, namespace, minioCtrName, minioCtrImage, spec.Port, envVars, int32(spec.Storage.NodeCount), spec.ObjectStoreDataDir)
+
+	nodeCount := int32(spec.Storage.NodeCount)
+	ss := v1beta2.StatefulSet{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    map[string]string{"app": minioLabel},
+		},
+		Spec: v1beta2.StatefulSetSpec{
+			Replicas: &nodeCount,
+			Selector: &meta_v1.LabelSelector{
+				MatchLabels: map[string]string{"app": minioLabel},
+			},
+			Template: podSpec,
+			VolumeClaimTemplates: []v1.PersistentVolumeClaim{
+				{
+					ObjectMeta: meta_v1.ObjectMeta{
+						Name:      minioVolumeName,
+						Namespace: namespace,
+					},
+					Spec: v1.PersistentVolumeClaimSpec{
+						AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceStorage: resource.MustParse(minioStorageGBString),
+							},
+						},
+					},
+				},
+			},
+			ServiceName: name,
+			// TODO: liveness probe
+		},
+	}
+
+	return appsClient.StatefulSets(namespace).Create(&ss)
+}
+
+func (c *MinioController) makeMinioService(name, namespace string, spec miniov1alpha1.ObjectStoreSpec) (*v1.Service, error) {
+	coreV1Client := c.context.Clientset.CoreV1()
+
+	svc, err := coreV1Client.Services(namespace).Create(&v1.Service{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    map[string]string{"app": minioLabel},
+		},
+		Spec: v1.ServiceSpec{
+			Type:     v1.ServiceTypeNodePort,
```

Comment:
Referring to my earlier comment, I think you'll want to create a service that is internal to the cluster and the admin will need to figure out how they want to expose externally in their environment. NodePort would just be for dev convenience

Reply:
Just to clarify, you're saying it's fine to deploy it in a way that it can't reach the outside world? If so, I'll just remove the Nodeport portion of the deployment code.